backend/services/mgmt/v1alpha1/connection-service/connection.go

```diff
@@ -3,136 +3,151 @@ package v1alpha1_connectionservice
+5. persist permissions in a separate tab (stretch)
+7. handle root user - since root user has access to everything, update to return checks if user is root
+*/
+
 func (s *Service) CheckConnectionConfig(
 	ctx context.Context,
 	req *connect.Request[mgmtv1alpha1.CheckConnectionConfigRequest],
 ) (*connect.Response[mgmtv1alpha1.CheckConnectionConfigResponse], error) {
 	logger := logger_interceptor.GetLoggerFromContextOrDefault(ctx)
 	connectionTimeout := uint32(5)
-	conn, err := s.sqlConnector.NewDbFromConnectionConfig(req.Msg.ConnectionConfig, &connectionTimeout, logger)
-	if err != nil {
-		return nil, err
-	}
 
-	db, err := conn.Open()
-	if err != nil {
-		return nil, err
-	}
-	defer func() {
-		if err := conn.Close(); err != nil {
-			logger.Error(fmt.Errorf("failed to close database connection: %w", err).Error())
+	switch req.Msg.ConnectionConfig.Config.(type) {
+	case *mgmtv1alpha1.ConnectionConfig_PgConfig:
+
+		var pgDbPrivilegeRows []*pg_queries.GetPostgresRolePermissionsRow
+		var role string
+		var privs []*mgmtv1alpha1.ConnectionRolePrivileges
+		schemaTablePrivsMap := make(map[string][]string)
+
+		conn, err := s.sqlConnector.NewPgPoolFromConnectionConfig(req.Msg.GetConnectionConfig().GetPgConfig(), &connectionTimeout, logger)
+
+		if err != nil {
+			return nil, err
 		}
-	}()
 
-	cctx, cancel := context.WithTimeout(ctx, 5*time.Second)
-	defer cancel()
-	// err = db.PingContext(cctx)
-	tx, err := db.BeginTx(cctx, &sql.TxOptions{ReadOnly: true})
-	if err != nil {
-		return nil, err
-	}
-	defer nucleusdb.HandleSqlRollback(tx, logger)
+		db, err := conn.Open(ctx)
+		if err != nil {
+			return nil, err
+		}
 
-	// queries the database to see if we can see the schemas and tables
-	// note that schemas with no tables will not appear in this query
-	pgQuery := `
-	SELECT 
-		nspname AS schema_name, 
-		COUNT(*) AS number_of_tables
-	FROM 
-		pg_catalog.pg_tables
-		JOIN pg_catalog.pg_namespace ON pg_namespace.nspname = pg_tables.schemaname
-	WHERE 
-		pg_namespace.oid NOT IN 
-			(SELECT oid FROM pg_catalog.pg_namespace WHERE nspname = 'pg_catalog' OR nspname = 'information_schema')
-	GROUP BY 
-		schema_name
-	ORDER BY 
-		number_of_tables DESC;
-	`
-	// note that schemas with no tables will not appear in this query
-	mysqlQuery := `
-SELECT 
-    TABLE_SCHEMA AS schema_name,
-    COUNT(*) AS number_of_tables
-FROM 
-    information_schema.tables 
-WHERE 
-    TABLE_SCHEMA NOT IN ('information_schema', 'mysql', 'performance_schema', 'sys')
-GROUP BY 
-    TABLE_SCHEMA
-ORDER BY 
-    number_of_tables DESC;
-`
-
-	var rows *sql.Rows
-	var hasRows bool // used to check if there are rows returned from the db
+		defer conn.Close()
 
-	switch req.Msg.ConnectionConfig.Config.(type) {
-	case *mgmtv1alpha1.ConnectionConfig_PgConfig:
-		rows, err = tx.QueryContext(cctx, pgQuery)
-	case *mgmtv1alpha1.ConnectionConfig_MysqlConfig:
-		rows, err = tx.QueryContext(cctx, mysqlQuery)
-	}
+		cctx, cancel := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
This 5s timeout might be enough for ping, but it may not be enough for a larger db that takes a while to get all of the row results back since it may be a heavier query.